src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -731,10 +756,20 @@ public void RegisterMessageHandler(Func<Message, CancellationToken, Task> handle
         protected virtual async Task<IList<Message>> OnReceiveAsync(int maxMessageCount, TimeSpan serverWaitTime)
         {
             ReceivingAmqpLink receiveLink = null;
+
+            if (this.isSessionReceiver)
```

Comment:
I thought of it initially that way. But then it would be confusing for someone reading MessageReceiver code to say why are we checking for SessionReceiver every time, so I left out the check for SessionReceiver here, so in one glance, people can know whether they should be interested in that or not..